assets/eip-5219/IDecentralizedApp.sol

```diff
@@ -0,0 +1,16 @@
+// SPDX-License-Identifier: CC0-1.0
+pragma solidity ^0.8.0;
+
+interface IDecentralizedApp {
+    /// @notice                     Send an HTTP-like request to this contract
+    /// @param  method              The HTTP method to use (e.g. GET, POST, PUT, DELETE)
+    /// @param  resource            The resource to request (e.g. "/asdf/1234" turns in to ["asdf", "1234"], and "/demo?qwerty=uiop" turns in to ["demo"])
+    /// @param  parameters          The parameters to send with the request (e.g. "querty=uiop&1234=5678" would be ["qwerty", "1234"])
+    /// @param  values              The values to send with the request (e.g. "querty=uiop&1234=5678" would be ["uiop", "5678"])
```

Comment:
Why not an array of tuples rather than two arrays?

Reply:
Huh, I'm not sure why I didn't think of that.